core/server/data/xml/sitemap/base-generator.js

```diff
@@ -55,13 +55,18 @@ _.extend(BaseSiteMapGenerator.prototype, {
         // Create all the url elements in JSON
         var self = this,
             nodes;
-        nodes = _.map(data, function (datum) {
+
+        nodes = _.reduce(data, function (nodeArr, datum) {
```

Comment:
filtering the data on `getData` makes sense and i would leave as is - it's a faster query to the database. 

my suggestion would be: move `validateDatum ` into `addOrUpdateUrl`.
then it would do: if there is new or updated data, we first check if it's validate based on conditions on the specific generator. if it's valid we call `createUrlNodeFromDatum`. That would mean `createUrlNodeFromDatum` never returns `false`. This function then has just one independent job: create a node.

But you can choose which solution you would like to go with: keep as is or take my suggestion.